python-backend/app/api/utils/models_mixins.py

```diff
@@ -55,15 +67,82 @@ def save(self, commit=True):
                 db.session.rollback()
                 raise e
 
+    def deep_update_from_dict(self, data_dict, depth=0):
```

Comment:
I would really like to see a pydoc description of this function. And maybe the core logic moved out into a seperate function file which is invoked in this class. Having 50 lines of library code in the Base db model class feels like it doesn't satisfy separation of concerns.